src/lib/db/event-store.ts

```diff
@@ -115,50 +116,44 @@ class EventStore extends EventEmitter implements IEventStore {
         }
     }
 
-    async getEventsFilterByType(name: string): Promise<IEvent[]> {
-        try {
-            const rows = await this.db
-                .select(EVENT_COLUMNS)
-                .from(TABLE)
-                .limit(100)
-                .where('type', name)
-                .andWhere(
-                    'id',
-                    '>=',
-                    this.db
-                        .select(this.db.raw('coalesce(max(id),0) as id'))
-                        .from(TABLE)
-                        .where({ type: DROP_FEATURES }),
-                )
-                .orderBy('created_at', 'desc');
-            return rows.map(this.rowToEvent);
-        } catch (err) {
-            this.logger.error(err);
-            return [];
+    async searchEvents(search: SearchEventsSchema = {}): Promise<IEvent[]> {
+        let q = this.db
+            .select(EVENT_COLUMNS)
+            .from(TABLE)
+            .limit(search.limit ?? 100)
+            .offset(search.offset ?? 0)
+            .orderBy('created_at', 'desc');
+
+        if (search.type) {
```

Comment:
Instead of these if statements for the specific where clauses you could consider building a object of the possible where constraints and use the "object" syntax for where in one go ([see knex docs](http://knexjs.org/guide/query-builder.html#where))

Reply:
Good point, though I think the current form works better for the nested where clause for the search query.

Added the `IEventTable` type at least, which should type-check the column names.